apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -0,0 +1,743 @@
+      '(\\:\\d+)?(\\/?[-a-z\\d%_.~+]*)*' +
+      '(\\?[;&a-z\\d%_.~+=-]*)?',
+  );
+
+  return pattern.test(url);
+};
+
+export const ApplicationDialog = ({
+  isOpen,
+  onClose,
+  isEdit,
+  selectedApplication,
+  currentReference,
+}: Props) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation(Translation.PromptBar);
+  const [name, setName] = useState('');
+  const [version, setVersion] = useState('');
+  const [versionHasFocus, setVersionHasFocus] = useState(false);
+  const [featuresDataHasFocus, setFeaturesDataHasFocus] = useState(false);
+  const [description, setDescription] = useState('');
+  const [deleteLogo, setDeleteLogo] = useState(false);
+  const [localLogoFile, setLocalLogoFile] = useState<string | undefined>();
+  const files = useAppSelector(FilesSelectors.selectFiles);
+  const featuresDataInputRef = useRef<HTMLTextAreaElement>(null);
+  const [features, setFeatures] = useState('');
+  const [featuresDataError, setFeaturesDataError] = useState('');
+  const [maxAttachments, setMaxAttachments] = useState(0);
+  const [completionUrl, setCompletionUrl] = useState('');
+  const [completionUrlError, setCompletionUrlError] = useState('');
+  const [submitted, setSubmitted] = useState(false);
+  const [nameError, setNameError] = useState<string | null>(null);
+  const [versionError, setVersionError] = useState<string | null>(null);
+  const [iconError] = useState<string | null>(null);
+  const nameInputRef = useRef<HTMLInputElement>(null);
+  const descriptionInputRef = useRef<HTMLTextAreaElement>(null);
+  const [filterParams, setFilterParams] = useState<string[]>([]);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [isPublishing, setIsPublishing] = useState(false);
+
+  const entity = selectedApplication && {
+    name: selectedApplication.name,
+    id: ApiUtils.decodeApiUrl(
+      selectedApplication.id ||
+        (selectedApplication as unknown as { application: string }).application,
+    ),
+    folderId: getFolderIdFromEntityId(selectedApplication.name),
+  };
+
+  const resetForm = () => {
+    setName('');
+    setVersion('');
+    setDescription('');
+    setFeatures('');
+    setMaxAttachments(0);
+    setCompletionUrl('');
+    setLocalLogoFile(undefined);
+    setDeleteLogo(false);
+    setFilterParams([]);
+  };
+
+  const loading = useAppSelector((state) => state.application.loading);
+
+  const getItemLabel = (item: string) => item;
+  const handleChangeFilterParams = useCallback((filterParams: string[]) => {
+    setFilterParams(filterParams);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    setSubmitted(false);
+    resetForm();
+    onClose(false);
+  }, [onClose]);
+
+  const nameOnBlurHandler = (e: FocusEvent<HTMLInputElement>) => {
+    const newName = prepareEntityName(e.target.value.trim(), {
+      forRenaming: true,
+    });
+    const pattern = /^[^!@#$^*()+]{2,160}$/;
+
+    if (!pattern.test(newName)) {
```

Comment:
I think it's redundant. We remove special chats inside prepareEntityName function. Also we have notAllowedSymbols variable which you can import